contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -2648,6 +2653,102 @@ rename_update_bbf_catalog(RenameStmt *stmt)
 	}
 }
 
+static void
+rename_object_update_bbf_schema_permission_catalog(RenameStmt *stmt, int rename_type)
+{
+	/* Update 'object_name' in 'babelfish_schema_permissions' */
+	Relation	bbf_schema_rel;
+	TupleDesc	bbf_schema_dsc;
+	ScanKeyData key[4];
+	HeapTuple	tuple_bbf_schema;
+	HeapTuple	new_tuple;
+	SysScanDesc scan;
+	Datum		new_record_bbf_schema[BBF_SCHEMA_PERMS_NUM_OF_COLS] = {0};
+	bool		new_record_nulls_bbf_schema[BBF_SCHEMA_PERMS_NUM_OF_COLS] = {false};
+	bool		new_record_repl_bbf_schema[BBF_SCHEMA_PERMS_NUM_OF_COLS] = {false};
+	const char *logical_schema_name;
+	const char *object_name;
+	char	*object_type = NULL;
+	int16		dbid = get_cur_db_id();
+	Node	   *schema;
+	ObjectWithArgs *objwargs;
+
+	/* open the catalog table */
+	bbf_schema_rel = table_open(get_bbf_schema_perms_oid(), RowExclusiveLock);
+	/* get the description of the table */
+	bbf_schema_dsc = RelationGetDescr(bbf_schema_rel);
+
+	if (rename_type == OBJECT_TABLE || rename_type == OBJECT_VIEW)
+	{
+		logical_schema_name = get_logical_schema_name(stmt->relation->schemaname, true);
+		object_name = stmt->relation->relname;
+		object_type = OBJ_RELATION;
+	}
+	else
+	{
+		if (rename_type == OBJECT_PROCEDURE)
+			object_type = OBJ_PROCEDURE;
+		else if (rename_type == OBJECT_FUNCTION)
+			object_type = OBJ_FUNCTION;
+		objwargs = (ObjectWithArgs *) stmt->object;
+		schema = (Node *) linitial(objwargs->objname);
+		logical_schema_name = strVal(schema);
+		object_name = stmt->subname;
+	}
```

Comment:
Is it mandatory to pass schema qualified name in sp_rename? If not then we won't be getting the schema at this point.

Reply:
Your question is not very clear to me. 
We are fetching the schema name and object name from the rename statement, based on which we search the catalog for any available entry and update it.

Reply:
For every test case of sp_rename, we are passing the schema qualified name with it, Can we include the test w/o schema name as well?

Reply:
1. Added tests for objects with schema name as well.
2. We are storing the object names in lower case in the catalog.

Reply:
I think Harsh is asking to add tests without schema name.

Reply:
Yes, added those tests already.